ocaml/libs/http-lib/http.ml

```diff
@@ -1018,27 +979,31 @@ module Url = struct
     let params = if params = [] then "" else "?" ^ kvpairs params in
     uri ^ params
 
-  let to_string = function
-    | File {path}, data ->
-        (* this should have file:// *)
-        Printf.sprintf "file:%s%s" path (data_to_string data)
-        (* XXX *)
-    | Http h, {uri; query_params= params} ->
-        let auth =
-          match h.auth with
-          | Some (Basic (username, password)) ->
-              Printf.sprintf "%s:%s" username password |> Option.some
-          | _ ->
-              Option.none
-        in
-        Uri.(
-          make
+  let to_string scheme =
```

Comment:
`scheme` here isn't of the same type as in `of_string` above, which is a little confusing. The type here is `type t = scheme * data`.